[Docs] Update change log with open issues and limitations (#4577)
mlrun/mlrun: docs/change-log/index.md

###  Closed issues
| ID     |Description                                                                   |
|---------|-------------------------------------------------------------------------------|
|ML-1584|Can now run `code_to_function` when filename contains special characters.|

> |ML-2199|Spark operator job doe not fail with default requests args.                   |

Reviewer: ```suggestion
|ML-2199|Spark operator job does not fail with default requests args.                   |
```

|ML-3508|Improved description of list_runs. See {py:class}`~mlrun.projects.MlrunProject.list_runs` [View in Git](https://github.com/mlrun/mlrun/pull/3686).                                                                                                                                                     |
|ML-3621|`clear_context()` now does not delete content if the path is relative; and if a subpath exists, only the sub dir is deleted/cleared. [View in Git](https://github.com/mlrun/mlrun/pull/3689).                                                                                                          |
|ML-3631|MLRun now successfully pulls the source code from gitlab with a personal access token. [View in Git](https://github.com/mlrun/mlrun/pull/3927).                                                                                                                                                        |

> |ML-3633|Can now import a context from dict. [View in Git](https://github.com/mlrun/mlrun/pull/3308).                                                                                                |

Reviewer: ```suggestion
|ML-3633|Fix date fields parsing when importing a context from dict. [View in Git](https://github.com/mlrun/mlrun/pull/3308).                                                                                                |
```

|ML-3508|Improved description of list_runs. See {py:class}`~mlrun.projects.MlrunProject.list_runs` [View in Git](https://github.com/mlrun/mlrun/pull/3686).                                                                                                                                                     |
|ML-3621|`clear_context()` now does not delete content if the path is relative; and if a subpath exists, only the sub dir is deleted/cleared. [View in Git](https://github.com/mlrun/mlrun/pull/3689).                                                                                                          |
|ML-3631 |MLRun now successfully pulls the source code from gitlab with a personal access token. [View in Git](https://github.com/mlrun/mlrun/pull/3927).                                                                                                                                                        |

> |ML-3633|Fail to import a context from dict. [View in Git](https://github.com/mlrun/mlrun/pull/3308).                                                                                                |

Reviewer 2: If this is a fix we should say it is now possible to do it

| ID| Description|Workaround |Opened in |
|--------|----------------------------------------------------------------|-----------------------------------------|-----------|

> |ML-1373|When attempting to ingest data with RemoteSpark using an incorrect service name, the pods get stuck during ContainerCreating. | NA | v0.9.0 | 

Reviewer 2: workaround: verify the correct service name

|ML-3294|Dask coredump during project deletion.|Before deleting a Dask project, verify that Dask was fully terminated.||v1.3.0 |
|ML-3315|Spark ingestion does not support nested aggregations.                 |NA |v1.2.1    |
|ML-3386|Documentation is missing full details on the feature store sources and targets.| NA|v1.2.1    |
|ML-3445|`project.deploy_function` operation might get stuck when running v1.3.0 demos on an Iguazio platform running v3.2.x.| Replace code: `serving_fn = mlrun.new_function("serving", image="python:3.9", kind="serving", requirements=["mlrun[complete]", "scikit-learn~=1.2.0"])` with: <br>`function = mlrun.new_function("serving", image="python:3.9", kind="serving") function.with_commands([ "python -m pip install --upgrade pip", "pip install 'mlrun[complete]' scikit-learn==1.1.2", ])`|v1.3.0    |
|ML-3480|Documentation: request details on label parameter of feature set definition.| NA | v1.2.1    |
|NA|The feature store does not support schema evolution and does not have schema enforcement.| NA| v1.2.1    |

> |ML-3633|Fail to import a context from dict                                     |When loading a context from dict (e.g.: mlrun.MLClientCtx.from_dict(context)), make sure to provide datetime objects and not string. Do this by executing `context['status']['start_time'] = parser.parse(context['status']['start_time'])<br> context['status']['last_update'] = parser.parse(context['status']['last_update'])` prior to `mlrun.MLClientCtx.from_dict(context)`|v1.3.0    |

Reviewer 2: The description here is too generic, need better explanation

Author: moved to fixed in v1.4.0

|ML-3640|When running a remote function/workflow, the `context` global parameter is not automatically injected.| Use `get_or_create_ctx`| v1.3.0    |
|ML-3714|It can happen that an MLrun pod succeeds but there's an error in Kubeflow. | NA | v1.3.0 |
|ML-4153|When creating a passthrough features-set in the UI, with no online target, the feature-set yaml includes a parquet offline target, which is ignored.| NA | v1.4.0  |

> |ML-4370| 

Reviewer 2: Missing info

|ML-4585|The `mlrun/mlrun` image does not support mpijob. | Create your own image that includes mpijob.  | v1.5.0 |
|ML-4613|UI: The Batch Inference Parameters has incorrect hint on map type. The correct hint is "The 'dict' values should be in JSON key:value format, e.g. {"hello":"world"}" | NA | v1.5.0 |
|ML-4740|When running function `batch_inference_v2` from the SDK, the `ingest()` function accepts 3 parameters as Data-item or other types: `dataset`, `model_path` and `model_endpoint_sample_set`. If you provided these parameters as non Data-items and later on you want to rerun this function from the UI, you need to provide these parameters as Data-item.|Prepare suitable Data-item and provide it to the batch-rerun UI.| v1.5.0    |
|ML-4758|In rare cases, deleting a heavy project is unsuccessful and results in a timeout error message while the project moves to offline state.| Delete again.| v1.5.0    |
|ML-4767|Torch 2.1.0 is not compatible with mlrun-gpu image.              | NA | v1.5.0 |
|ML-4907|MLRun Client does not support Win OS.                                                 | NA | v1.3.0 | 

> |ML-4942|Dask dashboard does not work on managed clouds.|Dask is externally available thru NodePort, meaning ports needs to be manually open by the infrastructure provider. If running MLRun locally or CE, make sure to port-forward the port Dask Dashboard uses to ensure it is available externally to the Kubernetes cluster. | v1.5.0 |

Reviewer 2: The description is too generic, the issue is that dask dashboard requires the relevant node ports to be open

> |ML-4992|Source target dir ownership is set to root. |Set target dir, and change workdir back: 

Reviewer 2: need more specific description for the issue

   )
   sj.spec.build.extra = "WORKDIR /igz"
   ```

## Limitations
| ID     |Description                                                                                                                                                                          |Workaround                                                                                                                                                                                                                                                                                                                      |Opened in |
|---------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------|     

> |ML-1278|A user's permission to store data on the underlying storage depends on its V3IO permissions, not on project membership.                                                               |NA                                                               | v0.8.0

Reviewer 2: When setting permissions to the project, the right permissions to the data storage need to be set in order to allow access to the data.

|ML-3206|When get_or_create_project is called, and there is project.yaml in the dir, no new project is created (even of the project name is new). The existing project.yaml is loaded instead.|v1.2.1                                                                                                                                                                                                                                                                                                                           |
|ML-3520|MLRun does not decompress large Kubeflow pipelines                                                                                                                                   |NA                                                                                                                                                                                                                                                                                                                              |v1.3.0    |

> |ML-3824|MLRun supports TensorFlow up to 2.11.                                                                                                                                                |NA                                                                                                                                                                                                                                                                                                                              |v1.3.1    |

Reviewer 2: Please recheck, I think that part was addressed

|ML-3731|When trying to identify a failed step in a workflow with `mlrun.get_run_db().list_pipelines('project-name')`, the returned error is `None`.                                          |To see the error, use `mlrun.db.get_pipelines()` instead.                                                                                                                                                                                                                                                                        |
|ML-3743|Setting AWS credentials as project secret cause a build failure on EKS configured with ECR.                                                                                          |When using an ECR as the external container registry, make sure that the project secrets AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY have read/write access to ECR, as described in the [platform documentation](https://www.iguazio.com/docs/latest-release/services/app-services/docker-registry/#create-off-cluster-registry)|

> |ML-4386|Local notifications aren’t persisted in mlrun API| NA | v1.5.0|

Reviewer 2: local runs with notifications

> |ML-4767|Use PyTorch versions up to and including than 2.0.1, but not higher.       | NA |                 v1.5.0|   

Reviewer 2: this is with the mlrun-gpu image, not general, but please recheck

> |ML-4960|The Memberships tab in Projects > Settings is part of MLRun only when integrated with the platform.            | NA |                 v1.5.0| 

Reviewer 2: The issue is with the caching, the fact that it's only part of the platform is not a limitation

|ML-4758|In rare cases, deleting a heavy project is unsuccessful and results in a timeout error message while the project moves to offline state.| Delete again.| v1.5.0    |
|ML-4767|Torch 2.1.0 is not compatible with mlrun-gpu image.              | NA | v1.5.0 |
|ML-4907|MLRun Client does not support Win OS.                                                 | Use WSL instead. | v1.3.0 | 
|ML-4942|The Dask dashboard requires the relevant node ports to be open. |Your infrastructure provider must open the ports manually. If running MLRun locally or CE, make sure to port-forward the port Dask Dashboard uses to ensure it is available externally to the Kubernetes cluster. | v1.5.0 |
|ML-4992|When a source archive is specified, the docker image's working directory is no longer automatically set to the target directory of that source archive.|Set target dir, and change workdir back: 
   ```sj.with_source_archive(
       source=project.source, pull_at_runtime=False, target_dir="/igz/mlrun_code"
   )
   sj.spec.build.extra = "WORKDIR /igz"

>    ```

Reviewer: ```suggestion
   ``` |v1.5.0|
```

## Limitations

> | ID     |Desc                                                                                                                                |Workaround |Opened in|

Reviewer: ?
```suggestion
| ID     |Description                                                                                                                                |Workaround |Opened in|
```

|ML-3743|Setting AWS credentials as project secret cause a build failure on EKS configured with ECR.                                                                                          |When using an ECR as the external container registry, make sure that the project secrets AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY have read/write access to ECR, as described in the [platform documentation](https://www.iguazio.com/docs/latest-release/services/app-services/docker-registry/#create-off-cluster-registry)|
|ML-4386|Notifications of local runs aren't persisted. | NA | v1.5.0|
|ML-4767|When using mlrun-gpu image, use PyTorch versions up to and including than 2.0.1, but not higher.       | NA |                 v1.5.0|   

> |ML-4855|MLRun supports TensorFlow up to 2.13.1.

Reviewer: this is duplicated with ML-3824
